diff -aur libmatecomponent-1.2.1/matecomponent-activation/matecomponent-activation-client.h libmatecomponent-1.2.1-patch/matecomponent-activation/matecomponent-activation-client.h
--- libmatecomponent-1.2.1/matecomponent-activation/matecomponent-activation-client.h	2012-03-27 13:52:14.000000000 -0500
+++ libmatecomponent-1.2.1-patch/matecomponent-activation/matecomponent-activation-client.h	2012-05-21 14:08:01.291378652 -0500
@@ -2,6 +2,10 @@
 #define MATECOMPONENT_ACTIVATION_CLIENT_H
 
 #include <glib.h>
+/* FIXME: redeclared, but should be removed */
+#ifdef G_DISABLE_DEPRECATED
+typedef void (*GVoidFunc) (void);
+#endif
 #include <matecomponent-activation-private.h>
 
 void matecomponent_activation_release_corba_client (void);
